scripts/pre_commit_linter.py

```diff
@@ -713,16 +718,151 @@ def _check_spacing(all_files):
     return summary_messages
 
 
+def _check_comments(all_files):
+    """This functions ensures that comments end in a period."""
+    print 'Starting comment checks'
+    print '----------------------------------------'
+    summary_messages = []
+    files_to_check = [
+        filename for filename in all_files if not
+        any(fnmatch.fnmatch(filename, pattern) for pattern in EXCLUDED_PATHS)
+        and filename.endswith('.py')]
+    message = 'There should be a period at the end of the comment.'
+    failed = False
+    for filename in files_to_check:
+        with open(filename, 'r') as f:
+            file_content = f.readlines()
+            file_length = len(file_content)
+            for line_num in range(file_length):
+                line = file_content[line_num].lstrip().rstrip()
+                next_line = ''
+                if line_num + 1 < file_length:
+                    next_line = file_content[line_num + 1].lstrip().rstrip()
+
+                if line.startswith('#') and not next_line.startswith('#'):
+                    # Check that the comment ends with the proper punctuation.
+                    if (
```

Comment:
if statement predicates should not have parenthesis around them, but sub expressions may. I would recommend doing something like:
```python
if (line[-1] not in
        ALLOWED_TERMINATING_PUNCTUATIONS) and (
           not any(word in line for word in EXCLUDED_PHRASES)):
```

Reply:
see for sample: https://github.com/oppia/oppia/blob/262920951b6f9b5ea6c493af32c5e9805f3df290/scripts/deploy.py#L75-L77